### ISSUE TITLE
Trying to destroy failed app deploy leads to crash 

### ISSUE THREAD

**Issue by unredundant** [digitalocean/terraform-provider-digitalocean#700]
# Bug Report

---

## Describe the bug
Hey 👋 Doing a bit of terraformin' on a beautiful sunday and ran into a relatively painless bug.  It appears that, should a `terraform apply` fail to successfully deploy a digital ocean app to the app platform, in the event that the user tries to then run a `terraform destroy` the operation results in some glorious golang puke 

### Expected Behavior
Destroy should not crash when a borked app is part of the manifest 

### Actual Behavior
Destroy does crash when a borked app is part of the manifest 

### Steps to Reproduce

1. Build a broken app
2. Try to deploy said broken app
3. Try to destroy said broken app

**Terraform Configuration Files**

Any broken app platform deploy should work

**Expected behavior**

No crashes :) 

**Debug Output**

```
Stack trace from the terraform-provider-digitalocean_v2.10.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xdce80e]

goroutine 37 [running]:
github.com/digitalocean/terraform-provider-digitalocean/digitalocean.resourceDigitalOceanAppRead(0x11f2ad8, 0xc000351f80, 0xc00066a980, 0xf28ec0, 0xc0005bac30, 0xc0007d0ca0, 0xc000707908, 0x40e0f8)
        github.com/digitalocean/terraform-provider-digitalocean/digitalocean/resource_digitalocean_app.go:111 +0x46e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0005d08c0, 0x11f2a68, 0xc0004b3c00, 0xc00066a980, 0xf28ec0, 0xc0005bac30, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:347 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0005d08c0, 0x11f2a68, 0xc0004b3c00, 0xc0006b47e0, 0xf28ec0, 0xc0005bac30, 0xc0007d8198, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000c030, 0x11f2a68, 0xc0004b3c00, 0xc0004b3c40, 0xc0004b3c00, 0x40b965, 0xfc8d60)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc00052b2c0, 0x11f2b10, 0xc0004b3c00, 0xc000350ea0, 0xc00052b2c0, 0xc000782780, 0xc000085ba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x1007700, 0xc00052b2c0, 0x11f2b10, 0xc000782780, 0xc000350e40, 0x0, 0x11f2b10, 0xc000782780, 0xc000788a00, 0x4e3)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000120000, 0x11fbfb8, 0xc0004a1680, 0xc000759600, 0xc0003a3110, 0x1769a90, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000120000, 0x11fbfb8, 0xc0004a1680, 0xc000759600, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00051a230, 0xc000120000, 0x11fbfb8, 0xc0004a1680, 0xc000759600)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-digitalocean_v2.10.1 plugin crashed!
```

**Panic Output**

## Additional context

**Important Factoids**

**References**

**Comment by opeco17** [comment]
Hi, @unredundant 

This issue was solved in v2.12.0 release
https://github.com/digitalocean/terraform-provider-digitalocean/blob/v2.12.0/digitalocean/resource_digitalocean_app.go#L122

You can use v2.12.0 or latter version to prevent this issue

**Comment by andrewsomething** [comment]
Thanks for pointing that out @opeco17. Looks like I forgot to close this when https://github.com/digitalocean/terraform-provider-digitalocean/pull/684 was merged.
